src/client/csfle/client_encryption.rs

```diff
@@ -271,42 +273,52 @@ pub struct DataKeyOptions {
 
 /// A KMS-specific key used to encrypt data keys.
 #[derive(Debug, Clone, Serialize)]
-#[serde(rename_all = "camelCase", untagged)]
+#[serde(untagged)]
 #[non_exhaustive]
 #[allow(missing_docs)]
 pub enum MasterKey {
+    #[serde(rename_all = "camelCase")]
     Aws {
         region: String,
         /// The Amazon Resource Name (ARN) to the AWS customer master key (CMK).
         key: String,
         /// An alternate host identifier to send KMS requests to. May include port number. Defaults
         /// to "kms.<region>.amazonaws.com"
+        #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Can these be replaced with a `#[serde_with::skip_serializing_none]` tag on the entire enum? Or does that not propagate to the nested fields?